model/task_queue_service.go

```diff
@@ -308,13 +328,53 @@ func (t *basicCachedDispatcherImpl) FindNextTask(spec TaskSpec) *TaskQueueItem {
 					"schedulableunit_running_hosts": unit.runningHosts,
 					"schedulableunit_max_hosts":     unit.maxHosts,
 					"schedulableunit_num_tasks":     len(unit.tasks),
-					"task_id_returned":              "",
 				})
 
 				return nil
 			}
 
 			// A non-task group schedulableUnit's tasks ([]TaskQueueItem) only contain a single element.
+			item := unit.tasks[0]
+
+			nextTaskFromDB, err := task.FindOneId(item.Id)
+			if err != nil {
+				grip.Error(message.WrapError(err, message.Fields{
```

Comment:
shouldn't be an error. 

can we just have a function in the task finder that returns a bool,err (for "is this ID met?")

Reply:
Standardising to `Error.Warning`

Reply:
Leaving as is per 1:1 discussion.